src/mjolnir/graphbuilder.cc

```diff
@@ -181,24 +182,40 @@ void ConstructEdges(const OSMData& osmdata,
             (way.link() && Length(edge.llindex_, way_node.node) < kMaxInternalLength);
         way_node.node.link_edge_ = way.link();
         way_node.node.non_link_edge_ = !way.link() && (way.auto_forward() || way.auto_backward());
-        nodes.push_back({way_node.node, static_cast<uint32_t>(-1),
-                         static_cast<uint32_t>(edges.size()), graph_id_predicate(way_node.node)});
+
+        uint32_t size = static_cast<uint32_t>(edges.size());
+        if (!edge.attributes.way_begin)
+          size += 1;
```

Comment:
what is this doing?

Reply:
can some one just describe in general the changes to this function. at a high level i cant quite grok them by looking at the code. like what do the changes accomplish for the sake of lane processing

Reply:
The changes are we need to flag the edge after the beginning and the edge prior to the last so that we can temporary set the internal flag.